fix: only update content-length header if one was present in spy (#25920)
cypress-io/cypress: cli/CHANGELOG.md

 - Fixed a compatibility issue so that component test projects can use [Vite](https://vitejs.dev/) version 4.2.0 and greater. Fixes [#26138](https://github.com/cypress-io/cypress/issues/26138).
 - Changed the way that Git hashes are loaded so that non-relevant runs are excluded from the Debug page. Fixes [#26058](https://github.com/cypress-io/cypress/issues/26058).

>  - Fixed an issue where using `cy.intercept` to spy re-computed a value for the `content-length` header even if no `content-length` header was present in the original request. Fixes [#24407](https://github.com/cypress-io/cypress/issues/24407).

Reviewer: Add link to `cy.intercept()` doc page

I'm also not sure reading this entry what is being fixed here. I think it needs re-worded a little. 
```suggestion
 - Fixed an issue where [`cy.intercept()`](https://docs.cypress.io/api/commands/intercept) added an additional `content-length` header to spied requests that did not set a `content-length` header on the original request. Fixes [#24407](https://github.com/cypress-io/cypress/issues/24407).
```

Reviewer 2: I committed the suggestion. It seems to read fairly well. I think the one thing I would change is changing `spied` to `intercepted`. What do you think?

Reviewer: OH good call out. That is the terminology used in the docs. https://docs.cypress.io/api/commands/intercept#Intercepted-requests

Author: One note is that I intentionally chose the language "spied" instead of "intercepted." This bug does not apply when `intercept` is used to stub, only when it is used to spy, so I chose the word "spied" to specify that it only relates to a subset of `intercept`'s functionality.

Reviewer: 🤔 okay -- totally fair. I'm good with spied. given you can do both stubbing or spying on intercepted requests.

Author: Aside from that I prefer the updated wording, thanks @emilyrohrbough

Reviewer: @Bourg Updated! 😄 thank you for your contribution!! we truly appreciate it!

>  - Fixed an issue where [`cy.intercept()`](https://docs.cypress.io/api/commands/intercept) added an additional `content-length` header to spied requests that did not set a `content-length` header on the original request. Fixes [#24407](https://github.com/cypress-io/cypress/issues/24407).

Reviewer 2: ```suggestion
 - Fixed an issue where [`cy.intercept()`](https://docs.cypress.io/api/commands/intercept) added an additional `content-length` header to intercepted requests that did not set a `content-length` header on the original request. Fixes [#24407](https://github.com/cypress-io/cypress/issues/24407).
```

>  - Fixed an issue where [`cy.intercept()`](https://docs.cypress.io/api/commands/intercept) added an additional `content-length` header to intercepted requests that did not set a `content-length` header on the original request. Fixes [#24407](https://github.com/cypress-io/cypress/issues/24407).

Reviewer: ```suggestion
 - Fixed an issue where [`cy.intercept()`](https://docs.cypress.io/api/commands/intercept) added an additional `content-length` header to spied requests that did not set a `content-length` header on the original request. Fixes [#24407](https://github.com/cypress-io/cypress/issues/24407).
```
